compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -840,10 +840,12 @@ extends Message(LossyWideningConstantConversionID):
                 |Write `.to$targetType` instead."""
   def explain(using Context) = ""
 
-class PatternMatchExhaustivity(uncoveredFn: => String, hasMore: Boolean)(using Context)
+class PatternMatchExhaustivity(val uncovered: Seq[String], tree: untpd.Match)(using Context)
 extends Message(PatternMatchExhaustivityID) {
   def kind = MessageKind.PatternMatchExhaustivity
-  lazy val uncovered = uncoveredFn
```

Comment:
You're changing the message from being the first 6 examples, comma-separated, to all examples rendered as a list (which is bracketed).

Reply:
Good catch, I forgot to restore `take(6)`. Fixed together with list rendering.